src/UI/V2/Dialog.elm

```diff
@@ -0,0 +1,119 @@
+module UI.V2.Dialog exposing
+    ( Dialog(..)
+    , dialogMap
+    , renderElement
+    , withBody
+    , withButtons
+    )
+
+import Element exposing (Element, fill, shrink)
+import Element.Border as Border
+import UI.Button as Button exposing (Button)
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.Colors exposing (mainBackground)
+import UI.Palette as Palette
+import UI.RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.Utils.Element as Element
+
+
+type Dialog msg
+    = Dialog Properties (Options msg)
+
+
+type alias Properties =
+    { title : String
+    , icon : Icon
+    }
+
+
+type alias Options msg =
+    { body : Element msg
+    , buttons : List (Button msg)
+    }
+
+
+dialogMap : (a -> b) -> Dialog a -> Dialog b
+dialogMap applier (Dialog { title, icon } { body, buttons }) =
+    Dialog
+        { title = title
+        , icon = icon
+        }
+        { body = Element.map applier body
+        , buttons = List.map (Button.map applier) buttons
+        }
```

Comment:
I'm failing to see when this function would be useful. When would we need to apply the same function to both the body and the buttons? Perhaps it makes sense to map the whole options record so that the user has the option to keep one property untouched